client/src/util/centroid.js

```diff
@@ -6,100 +6,131 @@ import { unassignedCategoryLabel } from "../globals";
   Centroid coordinate calculation
 */
 
-/* 
-  calcMedianCentroid goes through a given metadata category
-  fetches each cell's coordinates grouping by category value.
-
-  It then calculates the median value and puts that in the array
+/*
+Generates a mapping of categorical values to data needed to calculate centroids
+categoricalValue -> {
+ size: int,
+ holdsFinite: Boolean,
+ xCoordinates: Float32Array,
+ yCoordinates: F32 Array
+}
 */
-
-const calcMedianCentroid = (
+const getCoordinatesByCategoricalValues = (
   obsAnnotations,
   obsLayout,
   categoryName,
   layoutDimNames,
   categoricalSelection,
   schemaObsByName
 ) => {
+  const scratchPad = new Map();
```

Comment:
pick a better name - this name suggests that the contents are not results!

how about:  coordinatesByCategory?